### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
Hey Austin! Really sorry about how messy this submission is. I first tried fixing the commit message but then i broke my branch for a bit and then I forgot about the stretch goal so that's why there's so many commits!! All my changes are consolidated on the Files Changed tab though! Let me know if I need to fix anything. Thanks :)

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,40 +15,145 @@
...
+
+        //if unique, push random num into array of randomized indexes
+        randIndexes.push(index);
+        //also push the random element in array into shuffledArray
+        shuffledArray.push(array[index]);
+    }
+    return shuffledArray
+}
+
+//failed at first due to some typos. Got the randomization down but just needed to tweak the push mechanism since it was not pushing correctly. Finally got the test to pass
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:61]: Great work! Love the use of the while loop!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,40 +15,145 @@
...
 
 // b) Create the function that makes the test pass.
 
+//function minMax that takes in an array
+const minMax = (array) =>{
+    //sorts array in ascending order. basically if a-b returns less than 0 sort b in front of a
+    array.sort((a,b) => a-b)
+    //returns the first element and the last element of that array which would be the min and max since it has been sorted
+    return [array[0], array[array.length-1]]
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:91]: Yes!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,40 +15,145 @@
...
+
+    return noDupesArray
+}
+
+//took a couple tries. I actually had issues flattening doing something like the following:
+//let combinedArrays = [];
+//combinedArrays.push(...arr)
+//combinedArrays.flat()
+//realized .flat() returns a new array and needs to be assigned to a variable
+//test passes!
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:159]: Great work!

File: code_challenges.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
-# MINASWAN
+# MINASWAN - 'Matz is nice and so we are nice!'
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:2]: How do I give extra credit?

File: code_challenges.rb [hunk 5]
```diff
@@ -31,3 +47,15 @@
...
 # Expected output: 'Rotator is a palindrome'
+
+def palindrome string
+    if string.downcase == string.downcase.reverse
+        "#{string} is a palindrome"
+    else
+        "#{string} is not a palindrome"
+    end
+end
+
```
Comment (WalkerTexas-coder) [inline code_challenges.rb:58]: Excellent work!

File: interview-questions.md [hunk 6]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Your answer:
+  Your answer: props is short for properties. They are basically keys and values passed down from a parent component to a child component. They are referred to either as just props or functional props. To pass props from a parent to a child, you must add the prop to the child component call within the parent component as follows:
+  <Child
+    propName = {}
+  />
+  and then refer to it in the child component as {this.props.propName} .
 
-  Researched answer:
+  Researched answer: As above, props is referring to properties and are used to pass data between components. Specifically, they are used to pass information from parent to child. They are also read-only meaning that the data the parent is passing into the child component should not be changed by the child. We can either use props to pass in values or methods. They are considered a snapshot of a state.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:16]: Perfect!

File: interview-questions.md [hunk 7]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event is however the user interacts with the page. It just represents a change occuring in the web browser.
 
-  Researched answer:
+  Researched answer: DOM events are interactions from the user including but not limited to hovering or clicking on elements and typing inputs within forms. Events are referred to as the "firing of signals" to indicate that something is changing and an event is occurring.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:24]: Excellent!

File: interview-questions.md [hunk 8]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-  Your answer:
-
-  Researched answer:
+  Your answer: Object-orientated programming or OOP refers to a model of programming where everything you interact with is an object. This is in contrast with functional programming that usually just uses functions. 
 
+  Researched answer: As above, OOP is programming model that utilizes classes and objects. It was created to handle large and complex software systems and sets of changeable data. It allows us to create more general and reusable pieces of code. Functional programming uses functions, logic, and also avoids changing state. It follows the same principle of also being a reusable generic piece of code that changes output only if inputs change. Apparently some differences between the two are:
+  - For iteration, OOP uses loops and functional programming uses recursion.
+  - Functional Programming uses variables and functions while OOP uses objects and methods.
+  - OOP is best used when your operations are fixed and repetivive (adding new things rather than new processes)
+  - Functional is best used when you need many processes on a few things (adding new processes rather than new things)
```
Comment (WalkerTexas-coder) [inline interview-questions.md:36]: That is a really cool distinction that I hadn't clarified in my head before. Thanks!

File: interview-questions.md [hunk 9]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 4. What is the difference between a Float and an Integer in Ruby?
 
-  Your answer:
+  Your answer: A float is referring to a decimal and an Integer is referring to a whole number. You can force floats by including them in operators with Integers.
 
-  Researched answer:
+  Researched answer: Floats have decimals points while Integers are whole numbers.Integers can be positive, negative, or 0. Interestingly enough, Ruby will let you use underscores as commas when referencing large integers i.e. 1,000,000 -> 1_000_000. In reference to Floats, it seems there is a precision aspect. 'Float' is technically considered imprecise (i.e. 0.2 + 0.1 == 0.3 will return false because the result of those sum of numbers is technically 0.30000000000000004). We should use BigDecimal by wrapping each decimal as follows (BigDecimal("0.2") + BigDecimal("0.1") == 0.3) but important to note that it is 12x slower than just using Float.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:43]: https://www.youtube.com/watch?v=s9F8pu5KfyM

Comment (WalkerTexas-coder) [reply interview-questions.md:43]: great work!

File: interview-questions.md [hunk 10]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+  //just to note the decimal examples were found here: https://www.rubyguides.com/2016/07/numbers-in-ruby/ hopefully it is okay that I added some similar phrases from this resource since it is already pretty concise and has some interesting information.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:45]: Yea, this is a well known phenomenon and your example is a classic example. Kuddos!

**Review by WalkerTexas-coder — APPROVED** [review]
Rebecca, Amazing work! Everything looks top notch. You are totally correct the files changed page collates all changes so I don't really see the individual life cycle of the commits. Your Code challenges and ruby challenges are awesome!  And your Interview questions are exceptional! Honestly you did amazing work! Keep it up-- This PR is approved.

PS
If you don't already know about the tool "Set" (i.e. new Set() ) take a look into it. I bet you'll find it helpful and interesting.

**Pull request merged**